exporter/test_exporter.py

```diff
@@ -25,46 +29,156 @@ def client():
-
-        assert response.status_code == 500
+    assert mock_split_county.call_count == 0
+
+
+@mock.patch('main.export_split_county_tables')
+@mock.patch('google.cloud.bigquery.Client')
+def testExportDatasetTables_NoTables(
+    mock_bq_client: mock.MagicMock,
+    mock_split_county: mock.MagicMock,
+    client: FlaskClient
+):
+    # Set up mocks
+    mock_bq_instance = mock_bq_client.return_value
+    mock_bq_instance.list_tables.return_value = iter(())
+
+    payload = {
+        'dataset_name': 'my-dataset',
+        'demo_breakdown': 'age'
+    }
+    response = client.post('/', json=payload)
+
+    assert response.status_code == 500
+    assert mock_split_county.call_count == 0
+
+
+@mock.patch('main.export_split_county_tables')
+@mock.patch('google.cloud.bigquery.Client')
+def testExportDatasetTables_ExtractJobFailure(
+    mock_bq_client: mock.MagicMock,
+    mock_split_county: mock.MagicMock,
+    client: FlaskClient
+):
+    # Set up mocks
+    mock_bq_instance = mock_bq_client.return_value
+    mock_bq_instance.list_tables.return_value = test_tables
+    mock_extract_job = Mock()
+    mock_bq_instance.extract_table.return_value = mock_extract_job
+    mock_extract_job.result.side_effect = google.cloud.exceptions.InternalServerError(
+        'Internal')
+
+    payload = {
+        'dataset_name': 'my-dataset',
+        'demo_breakdown': 'age'
+    }
+    response = client.post('/', json=payload)
+
+    assert response.status_code == 500
+    assert mock_split_county.call_count == 1
+
+#
+#
+#
+
+
+# TEST ADDITIONAL COUNTY-LEVEL DATASET SPLIT FUNCTIONS
+_test_query_results_df = pd.DataFrame({
+    'county_fips': ["01001", "01002", "01003"],
+    'some_condition_per_100k': [None, 1, 2],
+})
+
+
+@mock.patch('main.export_nd_json_to_blob')
+@mock.patch('main.prepare_blob')
+@mock.patch('main.prepare_bucket')
+@mock.patch('main.get_query_results_as_df', return_value=_test_query_results_df)
+@mock.patch('google.cloud.bigquery.Client')
+def testExportSplitCountyTables(
+        mock_bq_client: mock.MagicMock,
+        mock_query_df: mock.MagicMock,
+        mock_prepare_bucket: mock.MagicMock,
+        mock_prepare_blob: mock.MagicMock,
+        mock_export: mock.MagicMock,
+        client: FlaskClient
+):
+
+    mock_bq_instance = mock_bq_client.return_value
+    mock_bq_instance.list_tables.return_value = test_tables
+
+    payload = {
+        'dataset_name': 'my-dataset',
+        'demo_breakdown': 'age'
+    }
+    client.post('/', json=payload)
+
+    # ensure initial call to bq client and county-level calls per state/terr
+    assert mock_bq_client.call_count == 1
+    assert mock_query_df.call_count == NUM_STATES_AND_TERRITORIES
+    assert mock_prepare_blob.call_count == NUM_STATES_AND_TERRITORIES
+    assert mock_export.call_count == NUM_STATES_AND_TERRITORIES
+
+    # ensure generated ndjson for bq.storage matches expected ndjson
+    generated_nd_json = mock_export.call_args[0][1]
+    assert (generated_nd_json ==
```

Comment:
I'm assuming we don't care about order here so lets sort these so we don't end up with annoying errors in the future. 
```
    assert (sorted(generated_nd_json) ==
 sorted(_test_query_results_df.to_json(orient="records",
                                           lines=True)))

Reply:
cool didt know sorted() was a thing